drivers/digital-io/max149x6/max14916.c

```diff
@@ -299,6 +299,9 @@ int max14916_init(struct max149x6_desc **desc,
 	return 0;
 
 gpio_err:
+	no_os_gpio_remove(descriptor->fault_gpio);
+	no_os_gpio_remove(descriptor->ready_gpio);
```

Comment:
for this change, you need to rework the conditional jumps . you don't want to call `no_os_gpio_remove` on a gpio descriptor before being even initialized.